MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/HistoryController.java

```diff
@@ -114,4 +122,36 @@ public String deleteAllRecordsFromHistory() {
         historyService.deleteAllRecordsFromHistory();
         return "emptyHistory";
     }
+
+    /**
+     * Handles requests for creating PDF file with histories which are displayed on the page
+     *
+     * @param request - HttpServletRequest
+     * @param response - HttpServletResponse
+     *
+     */
+    @RequestMapping(value="/getpdf", method= RequestMethod.GET)
+    public void getPDF(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        HttpSession session = request.getSession();
+        List<History> histories = (List) session.getAttribute("historiesSession");
+
+        final ServletContext servletContext = request.getSession().getServletContext();
+        final File tempDirectory = (File) servletContext.getAttribute("javax.servlet.context.tempdir");
+        final String temperotyFilePath = tempDirectory.getAbsolutePath();
+
+        String fileName = "History.pdf";
+        response.setContentType("application/pdf");
+        response.setHeader("Content-disposition", "attachment; filename="+ fileName);
+
+        try {
+            pdfCreatorService.createPDF(temperotyFilePath+"\\"+fileName, histories);
+            ByteArrayOutputStream baos = pdfCreatorService.convertPDFToByteArrayOutputStream(temperotyFilePath+"\\"+fileName);
+            OutputStream os = response.getOutputStream();
```

Comment:
ByteArrayOutputStream and OutputStream  implements AutoClosable, so declare them in the try with resources